### PR TITLE
Maintain status of self-deaf and self-mute for bot. 

### PR DESCRIPTION
**Pull request by BrandtHill**
Keep self_deaf and self_mute for reconnects.
Fix to_struct for voice event structs.

### CODE REVIEW

File: lib/nostrum/struct/event/voice_state.ex [hunk 1]
```diff
@@ -92,19 +92,19 @@ defmodule Nostrum.Struct.Event.VoiceState do
...
-      self_stream?: map["self_stream"],
-      self_video?: map["self_video"],
-      suppress?: map["suppress"],
-      request_to_speak_timestamp: convert_stamp(map["request_to_speak_timestamp"])
+      session_id: map.session_id,
+      deaf?: map.deaf,
+      mute?: map.mute,
+      self_deaf?: map.self_deaf,
+      self_mute?: map.self_mute,
+      self_stream?: map[:self_stream] || false,
```
Comment (jchristgit) [inline lib/nostrum/struct/event/voice_state.ex:104]: [I'm not even going to question why this is optional but the other ones isn't (on Discord's side).....](https://discord.com/developers/docs/resources/voice#voice-state-object)

File: lib/nostrum/struct/event/voice_state.ex [hunk 2]
```diff
@@ -92,19 +92,19 @@ defmodule Nostrum.Struct.Event.VoiceState do
   @doc false
   def to_struct(map) do
```
Comment (jchristgit) [inline lib/nostrum/struct/event/voice_state.ex:93]: Thanks so much for testing the dev version and fixing this. It seems my recent changes here were quite the mess 🍅

**Review by jchristgit — APPROVED** [review]
Rock solid. Thank you so much!

File: lib/nostrum/struct/event/voice_state.ex (hunk 1 above)
Comment (BrandtHill) [reply lib/nostrum/struct/event/voice_state.ex:104]: Yeah it's also super very cool that all the keys to this struct are atoms except for `"member"` which is a string for no reason 🤷‍♂️

Comment (jchristgit) [reply lib/nostrum/struct/event/voice_state.ex:104]: Consistency. You love to see it...

**Pull request merged**